corehq/apps/userreports/tests/test_indicators.py

```diff
@@ -438,3 +441,54 @@ def testDecimal(self):
         self._check_result(indicator, dict(foo=5.5), Decimal(5.5))
         self._check_result(indicator, dict(foo=None), None)
         self._check_result(indicator, dict(foo="banana"), None)
+
+
+class LedgerBalancesIndicatorTest(SimpleTestCase):
+    def setUp(self):
+        self.spec = {
+            "type": "ledger_balances",
+            "column_id": "soh",
+            "display_name": "Stock On Hand",
+            "ledger_section": "soh",
+            "product_codes": [
+                "abc",
+                "def",
+                "ghi",
+            ]
+        }
+        self.stock_states = {}
+        for val, case_id, product_id in [
+            (32, 'case1', 'abc'),
+            (85, 'case1', 'def'),
+            (11, 'case1', 'ghi'),
+        ]:
+            stock_state = self._make_stock_state(val, case_id, product_id)
+            self.stock_states[(case_id, product_id)] = stock_state
+
+    @staticmethod
+    def _make_stock_state(value, case_id, product_id, section_id='soh'):
+        return StockState(
+            stock_on_hand=value,
+            case_id=case_id,
+            product_id=product_id,
+            section_id=section_id,
+        )
+
+    def mock_getter(self, section_id, case_id, product_id):
+        if (case_id, product_id) in self.stock_states:
+            return self.stock_states[(case_id, product_id)]
+        else:
+            raise StockState.DoesNotExist()
+
+    @patch('corehq.apps.userreports.indicators.factory.StockState.objects')
```

Comment:
The patch is a bit too closely tied to the implementation for my liking (see also: current test failure), but I figure it's okay as long as the query isn't much more complicated than passing a couple parameters to `StockState.objects.get`